### PR TITLE
[4011] Add empty state to quick create

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/4011

### CODE REVIEW

File: ui/app/src/modules/Preview/QuickCreate.tsx [hunk 1]
```diff
@@ -160,6 +179,22 @@ export function QuickCreateMenu(props: QuickCreateMenuProps) {
             {item.label}
           </MenuItem>
         ))}
+        {
+          quickCreateContentList && quickCreateContentList.length === 0 &&
```
Comment (rart) [inline ui/app/src/modules/Preview/QuickCreate.tsx]: quickCreateContentList?.length

**Review by rart — CHANGES_REQUESTED** [review]
What's missing? Why still a draft?
Get "Learn More" link from:
- Check prior code
- Vita & Sumer

**Comment by jvega190** [comment]
Worked https://github.com/craftercms/craftercms/issues/3811 in this branch

File: ui/app/src/modules/Preview/QuickCreate.tsx [hunk 2]
```diff
@@ -134,10 +163,31 @@ export function QuickCreateMenu(props: QuickCreateMenuProps) {
...
+        }
       );
     }
-  }, [siteId]);
+  }, [siteId, dispatch]);
+
+  useEffect(() => {
+    getStudioInfo().subscribe(
+      (info) => {
+        setStudioVersion(info.version.packageVersion.substr(0, 3));
```
Comment (rart) [inline ui/app/src/modules/Preview/QuickCreate.tsx:180]: Could we get this from the freemarker template? If so we could put it in state/env and retrieve there from instead.

File: ui/app/src/services/content.ts [hunk 3]
```diff
@@ -1087,7 +1087,10 @@ export function getBulkUploadUrl(site: string, path: string): string {
 
 export function getQuickCreateContentList(siteId: string) {
   return get(`/studio/api/2/content/list_quick_create_content.json?siteId=${siteId}`).pipe(
-    pluck('response')
+    pluck('response'),
+    catchError((error: AjaxError) => {
+      throw error.response?.response ?? error;
```
Comment (rart) [inline ui/app/src/services/content.ts]: We’re not doing this consistently on API2 calls; there’s no real need to wrap API2 stuff. Remove the catch error/revert file.

File: ui/app/src/modules/Preview/QuickCreate.tsx (hunk 2 above)
Comment (rart) [inline ui/app/src/modules/Preview/QuickCreate.tsx]: Could we get this from the freemarker template? If so we could put it in state/env and retrieve there from instead.

File: ui/app/src/services/monitoring.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { get } from '../utils/ajax';
+import { catchError, pluck } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { AjaxError } from 'rxjs/ajax';
+
+export function getStudioInfo(): Observable<any> {
```
Comment (rart) [inline ui/app/src/services/monitoring.ts]: Type

File: ui/app/src/services/monitoring.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  return get('/studio/api/2/monitoring/version').pipe(
+    pluck('response'),
+    catchError((error: AjaxError) => {
```
Comment (rart) [inline ui/app/src/services/monitoring.ts]: Remove catch

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/models/Version.ts [hunk 6]
```diff
@@ -52,3 +52,16 @@ export interface VersionsStateProps {
...
+  javaVersion: string;
+  javaVm: string;
+  osArch: string;
+  osName: string;
+  osVersion: string;
+  packageBuild: string;
+  packageBuildDate: string;
+  packageName: string;
+  packageVersion: string;
+}
```
Comment (rart) [inline ui/app/src/models/Version.ts]: These file contains content item versions (unrelated). Revert and move to its own file.

File: ui/app/src/modules/Preview/QuickCreate.tsx (hunk 2 above)
Comment (rart) [reply ui/app/src/modules/Preview/QuickCreate.tsx]: @jdrossl is this info available to use on the FTL context? 👆🏻

**Review by rart — CHANGES_REQUESTED** [review]
.

File: ui/app/src/services/monitoring.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { get } from '../utils/ajax';
+import { pluck } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { StudioVersion } from '../models/Version';
+
+export function getStudioInfo(): Observable<StudioVersion> {
+  return get('/studio/api/2/monitoring/version').pipe(
```
Comment (rart) [inline ui/app/src/services/monitoring.ts:23]: @jdrossl is this info available to use on the FTL context? 👆🏻

Comment (jdrossl) [reply ui/app/src/services/monitoring.ts:23]: not directly in freemarker, you would need to do it in groovy and pass it

**Review by rart — APPROVED** [review]

**Pull request merged**